aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -150,36 +150,78 @@ public void ConfigureServices(IServiceCollection services)
+For more information, see the <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> clss in the API documentation.
 
 ## Service lifetimes
 
 Choose an appropriate lifetime for each registered service. ASP.NET Core services can be configured with the following lifetimes:
 
-**Transient**
+### Transient
 
-Transient lifetime services are created each time they're requested from the service container. This lifetime works best for lightweight, stateless services.
+Transient lifetime services (<xref:Microsoft.Extensions.DependencyInjection.ServiceCollectionServiceExtensions.AddTransient*>) are created each time they're requested from the service container. This lifetime works best for lightweight, stateless services.
 
-**Scoped**
+### Scoped
 
-Scoped lifetime services are created once per client request (connection).
+Scoped lifetime services (<xref:Microsoft.Extensions.DependencyInjection.ServiceCollectionServiceExtensions.AddScoped*>) are created once per client request (connection).
 
 > [!WARNING]
 > When using a scoped service in a middleware, inject the service into the `Invoke` or `InvokeAsync` method. Don't inject via constructor injection because it forces the service to behave like a singleton. For more information, see <xref:fundamentals/middleware/index>.
 
-**Singleton**
+### Singleton
 
-Singleton lifetime services are created the first time they're requested (or when `ConfigureServices` is run and an instance is specified with the service registration). Every subsequent request uses the same instance. If the app requires singleton behavior, allowing the service container to manage the service's lifetime is recommended. Don't implement the singleton design pattern and provide user code to manage the object's lifetime in the class.
+Singleton lifetime services (<xref:Microsoft.AspNet.OData.Builder.ODataModelBuilder.AddSingleton*>) are created the first time they're requested (or when `Startup.ConfigureServices` is run and an instance is specified with the service registration). Every subsequent request uses the same instance. If the app requires singleton behavior, allowing the service container to manage the service's lifetime is recommended. Don't implement the singleton design pattern and provide user code to manage the object's lifetime in the class.
 
 > [!WARNING]
 > It's dangerous to resolve a scoped service from a singleton. It may cause the service to have incorrect state when processing subsequent requests.
 
+## Service registration methods
+
+Each service registration extension method offers overloads that are useful in specific scenarios.
+
+| Method | Automatic<br>object<br>disposal | Mocks for<br>testing | Pass args |
+| ------ | :-----------------------------: | :------------------: | :-------: |
+| `Add{LIFETIME}<{INTERFACE}, {TYPE}>()`<br>Example:<br>`services.AddScoped<IMyDep, MyDep>();` | Yes | Yes | No |
```

Comment:
```suggestion
| `Add{LIFETIME}<{SERVICE}, {IMPLEMENTATION}>()`<br>Example:<br>`services.AddScoped<IMyDep, MyDep>();` | Yes | Yes | No |
```

Unfortunately this is used throughout the table so it'll take more than just accepting this suggestion :).

We avoid using "interface" for this in our APIs because it isn't required to be an interface. The service can also be identified by an `abstract` (or even non-`abstract`) class. All that is required by DI is that the *implementation type* must be assignable to a variable/field/etc. of the *service type* (i.e. it implements the interface, or inherits from the class).

Most other uses of "interface" are in examples that actually choose to use an interface which is fine. I'm mostly trying to avoid using that in places that refere to the general concept of a DI service, rather than a specific example. I'll do a quick pass and see if I can find other places where it would be worth updating our language.

And if this is a little too in-the-weeds for this doc, that's fine too, we can leave it.